src/basics.md

```diff
@@ -1144,6 +1145,74 @@ fn main() {
 }
 ```
 
+[ex-error-chain-avoid-discarding]: #ex-error-chain-avoid-discarding
+<a name="ex-error-chain-avoid-discarding"></a>
+## Avoid discarding errors during error conversions
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+[Matching] on different error types returned by a function is possible
+and relatively compact with [error-chain] crate. To do so,
+[`ErrorKind`] is used to determine the error type.
+
+To illustrate the error matching, a random integer generator web
+service will be queried via [reqwest] and then the response will be
+parsed. Errors can be generated by the Rust standard library,
+[reqwest] and by the web service. [`foreign_links`] are used for well
+defined Rust errors. The `errors` macro is used to create an
```

Comment:
`errors` is actually not a macro but a clause inside a `error_chain!` macro

Reply:
Thanks for the clarification! I did not knew that. Is it called a [DSL](https://rustbyexample.com/macros/dsl.html#domain-specific-languages-dsls)?

Reply:
> Is it called a DSL?

Yep, the error-chain crate defines a DSL via its macros.